src/coords.c

```diff
@@ -238,19 +237,30 @@ npy_intp CountCoords(PyArrayObject *arr)
     if (iternext == NULL) { NpyIter_Deallocate(iter); return -1; }
     dataptr = NpyIter_GetDataPtrArray(iter);
 
+    GEOS_INIT;
+
     do {
         /* get the geometry */
         obj = *(GeometryObject **) dataptr[0];
-        if (!get_geom(obj, &geom)) { result = -1; goto finish; }
+        if (!get_geom(obj, &geom)) {
+            errstate = PGERR_NOT_A_GEOMETRY;
+            result = -1;
+            goto finish;
+        }
         /* skip incase obj was None */
         if (geom == NULL) { continue; }
         /* count coordinates */
-        ret = GEOSGetNumCoordinates_r(context, geom);
-        if (ret < 0) { result = -1; goto finish; }
+        ret = GEOSGetNumCoordinates_r(ctx, geom);
+        if (ret < 0) {
+            errstate = PGERR_GEOS_EXCEPTION;
+            result = -1;
+            goto finish;
+        }
         result += ret;
     } while(iternext(iter));
 
     finish:
+    GEOS_FINISH;
     NpyIter_Deallocate(iter);
     return result;
```

Comment:
Another small style nit: in other places we indent the part after `finish: ` ?

Reply:
Although in other files we actually also don't do this consistently, so no matter